packages/block-editor/src/components/block-settings-menu/block-settings-dropdown.js

```diff
@@ -174,6 +191,27 @@ export function BlockSettingsDropdown( {
 	const parentBlockIsSelected =
 		selectedBlockClientIds?.includes( firstParentClientId );
 
+	// When a currentClientId is in use, treat the menu as a controlled component.
+	// This ensures that only one block settings menu is open at a time.
+	const open = ! currentClientId
+		? undefined
+		: openedBlockSettingsMenu === currentClientId || false;
+
+	const onToggle = useCallback(
+		( localOpen ) => {
+			if ( localOpen && openedBlockSettingsMenu !== currentClientId ) {
+				setOpenedBlockSettingsMenu( currentClientId );
+			} else if (
+				! localOpen &&
+				openedBlockSettingsMenu &&
```

Comment:
Is `openedBlockSettingsMenu &&` required or would `openedBlockSettingsMenu === currentClientId` take care of the check anyway?

Reply:
This is a little subtle, but because `currentClientId` isn't required for this component, we need to check for truthiness because it's possible for both values to be `undefined`. It's a tiny optimisation, but this means that if everything is `undefined` we don't perform a call to set the `openedBlockSettingsMenu` state to be `undefined`.

Reply:
> it's possible for both values to be undefined

Oh, that explains it. Thank you!